Automattic/wp-calypso
# LinkInBio Flow: Plan step to use new grid

Related to Automattic/martech#1533

## Proposed Changes

Updates the Link In Bio flow to use the new 2023 grid with expected features and copy. See: https://github.com/Automattic/martech/issues/1533#issuecomment-1495798000 

This is based on https://github.com/Automattic/wp-calypso/pull/75174 and will need a rebase. The plan type selector may appear broken, but it will be fixed in the base branch once updated.

### Media

<img width="700" alt="Screenshot 2023-04-27 at 2 09 13 PM" src="https://user-images.githubusercontent.com/1705499/234882410-6a26880b-bb3a-4e45-8995-aec684b41a51.png">

## Testing Instructions



1. Go to `/setup/link-in-bio/plans`
2. Verify the new grid matches the media and mocks. Confirm functionality checks out
3. Go to `/start/plans` and `/plans/[SITE]` and ensure no regressions

## Pre-merge Checklist
